e2e/transactions/transactions_test.go

```diff
@@ -764,36 +783,56 @@ func (s *TransactionsTestSuite) TestEvictionOfQueuedTransactions() {
 	s.NoError(s.Backend.AccountManager().SelectAccount(TestConfig.Account1.Address, TestConfig.Account1.Password))
 
 	txQueue := s.Backend.TxQueueManager().TransactionQueue()
-	var i = 0
+
+	var i = int32(0)
 	txIDs := [txqueue.DefaultTxQueueCap + 5 + 10]common.QueuedTxID{}
+	txQueueMutex := sync.RWMutex{}
 	s.Backend.TxQueueManager().SetTransactionQueueHandler(func(queuedTx *common.QueuedTx) {
-		log.Info("tx enqueued", "i", i+1, "queue size", txQueue.Count(), "id", queuedTx.ID)
-		txIDs[i] = queuedTx.ID
-		i++
+		n := atomic.LoadInt32(&i)
+		log.Info("tx enqueued", "i", n+1, "queue size", txQueue.Count(), "id", queuedTx.ID())
+		txQueueMutex.Lock()
+		txIDs[n] = queuedTx.ID()
+		txQueueMutex.Unlock()
+
+		atomic.AddInt32(&i, 1)
 	})
 
 	s.Zero(txQueue.Count(), "transaction count should be zero")
 
+	wg := sync.WaitGroup{}
 	for j := 0; j < 10; j++ {
-		go s.Backend.SendTransaction(context.TODO(), common.SendTxArgs{}) // nolint: errcheck
+		wg.Add(1)
+		go func() {
+			s.Backend.SendTransaction(context.TODO(), common.SendTxArgs{}) // nolint: errcheck
+			wg.Done()
+		}()
 	}
 	time.Sleep(2 * time.Second) // FIXME(tiabc): more reliable synchronization to ensure all transactions are enqueued
 
+	txQueueMutex.Lock()
 	log.Info(fmt.Sprintf("Number of transactions queued: %d. Queue size (shouldn't be more than %d): %d",
-		i, txqueue.DefaultTxQueueCap, txQueue.Count()))
+		atomic.LoadInt32(&i), txqueue.DefaultTxQueueCap, txQueue.Count()))
+	txQueueMutex.Unlock()
 
 	s.Equal(10, txQueue.Count(), "transaction count should be 10")
 
-	for i := 0; i < txqueue.DefaultTxQueueCap+5; i++ { // stress test by hitting with lots of goroutines
-		go s.Backend.SendTransaction(context.TODO(), common.SendTxArgs{}) // nolint: errcheck
+	for j := 0; j < txqueue.DefaultTxQueueCap+5; j++ { // stress test by hitting with lots of goroutines
+		wg.Add(1)
+		go func() {
+			s.Backend.SendTransaction(context.TODO(), common.SendTxArgs{}) // nolint: errcheck
+			wg.Done()
+		}()
 	}
 	time.Sleep(5 * time.Second)
```

Comment:
I think you could remove this sleep

Reply:
We couldnt we need to wait blockchain.

Reply:
Ok

Reply:
Is there a way like at https://github.com/status-im/status-go/blob/develop/testing/testing.go#L88 ?